tfhe/src/integer/ciphertext/compact_list.rs

```diff
@@ -324,9 +445,87 @@ impl ParameterSetConformant for CompactCiphertextList {
 pub const WRONG_UNPACKING_MODE_ERR_MSG: &str =
     "Cannot expand a CompactCiphertextList that requires unpacking without \
     a server key, please provide a integer::ServerKey passing it with the \
-    enum variant IntegerCompactCiphertextListUnpackingMode::UnpackIfNecessary \
+    enum variant IntegerCompactCiphertextListExpansionMode::UnpackIfNecessary \
+    or IntegerCompactCiphertextListExpansionMode::CastAndUnpackIfNecessary \
     as unpacking_mode.";
 
+struct IntegerUnpackingToShortintCastingModeHelper {
+    msg_extract: Box<dyn Fn(u64) -> u64 + Sync>,
+    carry_extract: Box<dyn Fn(u64) -> u64 + Sync>,
+    msg_extract_bool: Box<dyn Fn(u64) -> u64 + Sync>,
+    carry_extract_bool: Box<dyn Fn(u64) -> u64 + Sync>,
+}
+
+impl IntegerUnpackingToShortintCastingModeHelper {
+    pub fn new(ksk_view: KeySwitchingKeyView<'_>) -> Self {
```

Comment:
Maybe this could  take a &ServerKey isntead and be used in `unpack_and_sanitize_message_and_carries` to share lut definitions

Reply:
indeed

Reply:
hmm actually, I don't pass luts to shortint as there are already hard coded functions in the casting (to make casting work) so I don't generate the LUTs but rather pass callbacks, what did you have in mind here ?

Reply:
ah right for the unpack to call into the function to generate the call backs